src/validations/validations.ts

```diff
@@ -44,96 +73,112 @@ export function validateMetatile(metatile: number): void {
 }
 
 /**
- * Validates that the input `scaleSet` is valid
- * @param scaleSet the scale set to validate
- */
-export function validateScaleSet(scaleSet: ScaleSet): void {
-  const arr = [...scaleSet.scaleDenominators.entries()];
-  for (let i = 0; i < arr.length - 1; i++) {
-    if (arr[i][0] + 1 !== arr[i + 1][0]) {
-      throw new Error("scale set must have it's zoom levels ordered in ascending order and must be larger then the previous by 1");
-    }
-
-    if (arr[i][1] <= arr[i + 1][1]) {
-      throw new Error("scale set must have it's scales ordered in ascending order and must be larger then the previous");
-    }
-  }
-}
-
-/**
- * Validates that the input `tileGrid` is valid
- * @param tileGrid the tile grid to validate
+ * Validates that the input `tileMatrix` is valid
+ * @param tileMatrix the tile matrix to validate
  */
-export function validateTileGrid(tileGrid: TileGrid): void {
-  validateBoundingBox(tileGrid.boundingBox);
-  validateScaleSet(tileGrid.wellKnownScaleSet);
+export function validateTileMatrix(tileMatrix: TileMatrix): void {
+  // validateBoundingBox(tileMatrix.boundingBox);
+  // validateScaleSet(tileMatrix.wellKnownScaleSet);
 
-  if (tileGrid.numberOfMinLevelTilesX < 1) {
-    throw new Error('number of tiles on the x axis of a tile grid at the min zoom level must be at least 1');
+  if (tileMatrix.matrixWidth < 1) {
+    throw new Error('width of tile matrix must be at least 1');
   }
 
-  if (tileGrid.numberOfMinLevelTilesY < 1) {
-    throw new Error('number of tiles on the y axis of a tile grid at the min zoom level must be at least 1');
+  if (tileMatrix.matrixHeight < 1) {
+    throw new Error('height of tile matrix must be at least 1');
   }
 
-  if (tileGrid.tileWidth < 1) {
-    throw new Error('tile width of a tile grid must be at least 1');
+  if (tileMatrix.tileWidth < 1) {
+    throw new Error('tile width of a tile matrix must be at least 1');
   }
 
-  if (tileGrid.tileHeight < 1) {
-    throw new Error('tile height of a tile grid must be at least 1');
+  if (tileMatrix.tileHeight < 1) {
+    throw new Error('tile height of a tile matrix must be at least 1');
   }
 }
 
 /**
- * Validates that the input `boundingBox` is a valid bounding box inside the tile grid's bounding box
- * @param boundingBox the bounding box to validate
- * @param referenceTileGrid the tile grid to validate the `boundingBox` against its own bounding box
+ * Validates that the input `boundingBox` is a valid bounding box inside the tile matrix with respect to `tileMatrix`
```

Comment:
`is a valid bounding box with respect to 'tileMatrix'`?

Reply:
changed